lib/analyser.py

```diff
@@ -3,3 +3,134 @@
 # Given a stock, return whether it should be considered for manual review
 def analyse(stock: Stock) -> bool:
     raise Exception("Not implemented")
+
+def calc_NPV(self, init_profit, init_profit_growth_rate, perpetual_growth_rate, \
+            years_until_decay, discount_rate, init_fixed_asset, market_cap, no_of_years):
+    """
+    Params:
+    1. Calculate the Net Present Value depending upon csh flows
+    2. init_profit = {CFO, (OP - Tax paid) (this is better)}
+    3. init_profit_growth_rate = can be list of gaussian distributions with 3 different means
+                                    The center mean can be an average of the last 3 years while 
+                                    upper and lower can be +/-4%
+    4. Perpetual growth rate = can be estimate of inflation
+    5. years_until_decay = time till which company would grow at a rate higher than perpetual
+                            growth rate
+    6. discount_rate = assume a high discount rate
+    7. init_fixed_asset is used in capex estimate
+        7.1. Use the New fixed asset = PAT/(fixed Asset*new PAT)
+                                        = fixed_asset*(1+profit_growth_rate)
+        7.2. growth capex = (New fixed asset - fixed asset)
+        7.3. Add maintainence capex of 4% of fixed assets             
+        7.4. capex = (New fixed asset - fixed asset) + 0.04*fixed asset
+        7.5. update the fixed asset with new capex
+    8. market cap
+    9. no of years = real holding time of the investment
+    """
+    # Initializations
+    net_present_value   = 0
+    profit              = init_profit       #base profit
+    fixed_asset         = init_fixed_asset  #base fixed asset
+    capex               = 0
+
+    # Linear decay factor 
+    decay_factor = (init_profit_growth_rate - perpetual_growth_rate)/years_until_decay
+
+    for year in range(0, no_of_years):
+        #Calculate the profit growth & capex for each year
+        if(year<years_until_decay):
+            pgr = init_profit_growth_rate - (decay_factor*year)
+            capex = (fixed_asset*(pgr - perpetual_growth_rate + 0.04))
+        else:
+            pgr = perpetual_growth_rate
+            capex = fixed_asset*0.04
+        
+        #calculate the profit and the fixed asset value
+        profit = profit*(1+pgr)
+        fixed_asset = fixed_asset + capex
+        #Calculate the discounted value of cash after capex spending
+        discounted_profit = (profit-capex)/(pow((1+discount_rate),(year+1)))
+        net_present_value += discounted_profit
+
+    terminal_value = profit*(1+perpetual_growth_rate)/(discount_rate-perpetual_growth_rate)
+    discounted_terminal_value = terminal_value/(pow((1+discount_rate),(year+1)))
+    net_present_value += discounted_terminal_value
+    return (market_cap/net_present_value)
+
+def DCF_MC_analysis(self, init_profit, init_fixed_asset, market_cap):
```

Comment:
```suggestion
def dcf_mc_analysis(self, init_profit, init_fixed_asset, market_cap):
```

Reply:
Followed the standard convention and implemented the changes